fairlearn/datasets/synthetic.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+import numpy as np
+from sklearn.datasets import make_classification
+
+
+def make_synthetic_dataset(seed=12345):
```

Comment:
Is it normal to specify a seed as default? I would have assumed that it uses something like system time by default.

Reply:
I'm not sure what's more common, but I think using `seed=None` here is quite reasonable. I'll make the change.

Reply:
Actually, I stand corrected. https://scikit-learn.org/stable/modules/generated/sklearn.datasets.make_classification.html?highlight=make_classification#sklearn.datasets.make_classification
We should probably do what sklearn does and use `random_state`. We can probably reuse their `check_random_state` code from https://github.com/scikit-learn/scikit-learn/blob/2beed55847ee70d363bdbfe14ee4401438fba057/sklearn/datasets/_samples_generator.py#L159 since it's part of `sklearn.utils` (which we already heavily rely on)

Reply:
Just saw this comment... I didn't know about `check_random_state`. Very useful! I've incorporated it now.